typescript/t.d.ts

```diff
@@ -116,8 +116,8 @@ type ParseKeysByFallbackNs<Keys extends $Dictionary> = _FallbackNamespace extend
   ? Keys[UnionFallbackNs]
   : Keys[_FallbackNamespace & string];
 
-type FilterKeysByContext<Keys, TOpt extends TOptions> = TOpt['context'] extends string
-  ? Keys extends `${infer Prefix}${_ContextSeparator}${TOpt['context']}${infer Suffix}`
+type FilterKeysByContext<Keys, Context extends any> = Context extends string
+  ? Keys extends `${infer Prefix}${_ContextSeparator}${Context}${infer Suffix}`
     ? `${Prefix}${Suffix}`
     : never
   : Keys;
```

Comment:
`FilterKeysByContext` is only used by `ParseKeys` which already check that `Context`  value is a `string` before  so it can be changed to something like this:

```ts
type FilterKeysByContext<Keys, Context extends string> = 
  ? Keys extends `${infer Prefix}${_ContextSeparator}${Context}${infer Suffix}`
    ? `${Prefix}${Suffix}`
    : never;
```

With this change we can also remove `extends any` which doesn't add any type specifity

Reply:
@marcalexiei thanks, actually the most important part is the `const ...` so I improved the fix further so that the code is more readable (amended the commit sorry).

Reply:
I applied code formatting which changed other places in the file, not sure if it's legit or not, let me know :)

Reply:
> actually the most important part is the const ... so I improved the fix further so that the code is more readable

👌

---

> I applied code formatting which changed other places in the file, not sure if it's legit or not, let me know :)

Strange, code base  already follows formatting rules.
Actually these changes do not follow `prettier` styles so related PR check is failing.
Try running:

```sh
npm run format:fix
```

Reply:
done thanks, for some reason my IDE wouldn't pick up the proper prettier config...